QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,190 @@
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rmw_implementation/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rmw_implementation/)
+
+### Documentation Policy [2.v]
+
+`rmw_implementation` does not expose a public API. There is no need to add documentation to new changes. In the case that this package has a public API the documetation of new functionality will be required.
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+`rmw_implementation` does not have any feature documentation and it will need to be added for higher quality levels.
+
+### Public API Documentation [3.ii]
+
+`rmw_implementation` does not expose a public API.
+
+### License [3.iii]
+
+The license for `rmw_implementation` is Apache 2.0, and a summary is in each source file, the type is declared in the [package.xml](./package.xml) manifest file, and a full copy of the license is in the [LICENSE](./LICENSE) file.
+
+There is an automated test which runs a linter (ament_copyright) that ensures each file has a license statement.
+
+### Copyright Statement [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `rmw_implementation`.
+
+There is an automated test which runs a linter (ament_copyright) that ensures each file has at least one copyright statement.
+
+Most recent test results can be found [here](http://build.ros2.org/view/Epr/job/Epr__class_loader__ubuntu_bionic_amd64/lastBuild/testReport/rmw_implementation/)
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+There are currently no public features undergoing tests.
+
+### Public API Testing [4.ii]
+
+`rmw_implementation` does not expose a public API. There are currently no tests for the public API.
+
+### Coverage [4.iii]
+
+`rmw_implementation` does not currently track test coverage.
+
+### Performance [4.iv]
+
+`rmw_implementation` does not currently have performance tests.
+
+### Linters and Static Analysis [4.v]
+
+`rmw_implementation` uses and passes all the standard linters and static analysis tools for a C++ package as described in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#linters-and-static-analysis).
+
+Currently nightly results can be seen here:
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/rmw_implementation/)
+* [linux-arm64_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rmw_implementation/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rmw_implementation/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rmw_implementation/)
+
+## Dependencies [5]
+
+### Direct Runtime ROS Dependencies [5.i/5.ii]
```

Comment:
None of these dependencies is "optional" right? might be a good idea stating in 5.ii if that's the case.